debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSource.java

```diff
@@ -209,11 +210,15 @@ private void readOplog(MongoClient primary, MongoPrimary primaryClient, ReplicaS
             filter = Filters.and(filter, operationFilter);
         }
 
-        final FindIterable<Document> results = oplog.find(filter)
+        FindIterable<Document> results = oplog.find(filter)
                 .sort(new Document("$natural", 1))
                 .oplogReplay(true)
                 .cursorType(CursorType.TailableAwait);
 
+        if (connectorConfig.getCursorMaxAwaitTime() > 0) {
+            results = results.maxAwaitTime(connectorConfig.getCursorMaxAwaitTime(), TimeUnit.MILLISECONDS);
```

Comment:
I wonder if it makes sense to call `noCursorTimeout` as well when this config option would be set.

Reply:
We could but to be clear my understanding is that `noCursorTimeout` only prevents idle cursors from being closed by the server automatically.  This is very different from the actual reported issue; however the stack trace I put in the comments on the issue I believe would be solved by this extra addition.